Feat/do 3489/cmitzel

cmitzel-ncino opened the pull request:
This pull request introduces several enhancements and refinements to GitHub Actions workflows, focusing on publishing processes, semantic commit linting, and CI/CD pipeline improvements. The most notable changes include the addition of new workflows for publishing releases and packages, updates to existing workflows to improve functionality, and consistency improvements in YAML formatting.

### New workflows for publishing:

* [`.github/workflows/publish.yml`](diffhunk://#diff-551d1fcf87f78cc3bc18a7b332a4dc5d8773a512062df881c5aba28a6f5c48d7R1-R60): Added a new workflow for publishing GitHub releases, including steps for repository setup, semantic release, and release summary generation.
* [`workflow-templates/1gp-publish.yml`](diffhunk://#diff-47c5453c8f788e138e59e77f9ea8ca7a10911931f27c842102a546a75485268eR1-R295): Introduced a detailed workflow for publishing Salesforce 1GP packages, covering commit linting, package deployment, and Jira ticket transitions.
* [`workflow-templates/1gp-publish.properties.json`](diffhunk://#diff-b124fb281bcc5e8f9848d54b8ebc611a60e28c1d48bf2357623469ac123f2649R1-R7): Added metadata for the 1GP publish workflow, including name, description, and categories.

### Updates to existing workflows:

* [`workflow-templates/1gp-ci.yml`](diffhunk://#diff-91bc36802f73c95624da2b6d3c2f86e2bd67f6008a91c7dd121f778da5ad4fbbL3-R9): Enhanced CI workflow by adding support for `merge_group` event types, updating concurrency settings, and improving YAML formatting for consistency. [[1]](diffhunk://#diff-91bc36802f73c95624da2b6d3c2f86e2bd67f6008a91c7dd121f778da5ad4fbbL3-R9) [[2]](diffhunk://#diff-91bc36802f73c95624da2b6d3c2f86e2bd67f6008a91c7dd121f778da5ad4fbbL59-R68) [[3]](diffhunk://#diff-91bc36802f73c95624da2b6d3c2f86e2bd67f6008a91c7dd121f778da5ad4fbbL95-R93) [[4]](diffhunk://#diff-91bc36802f73c95624da2b6d3c2f86e2bd67f6008a91c7dd121f778da5ad4fbbL183-R181) [[5]](diffhunk://#diff-91bc36802f73c95624da2b6d3c2f86e2bd67f6008a91c7dd121f778da5ad4fbbL219-R217)
* [`workflow-templates/commit-lint.yml`](diffhunk://#diff-00792edf54091733acbd3b5bfe232ade8e20e644e2cc675977bd44e9ecf569eeL3-R4): Modified the commit lint workflow to support `merge_group` event types and updated descriptions for semantic commit detection. [[1]](diffhunk://#diff-00792edf54091733acbd3b5bfe232ade8e20e644e2cc675977bd44e9ecf569eeL3-R4) [[2]](diffhunk://#diff-00792edf54091733acbd3b5bfe232ade8e20e644e2cc675977bd44e9ecf569eeL19-R18)

### Formatting consistency improvements:

* [`workflow-templates/hybrid-ci.yml`](diffhunk://#diff-d43d707a175a4a7bdccf0c203e1029a2a93cc2fc8e375c5a056b1d840ab50ddcL151-R151): Standardized the formatting of boolean values and strings by using single quotes for consistency across the file. [[1]](diffhunk://#diff-d43d707a175a4a7bdccf0c203e1029a2a93cc2fc8e375c5a056b1d840ab50ddcL151-R151) [[2]](diffhunk://#diff-d43d707a175a4a7bdccf0c203e1029a2a93cc2fc8e375c5a056b1d840ab50ddcL180-R183) [[3]](diffhunk://#diff-d43d707a175a4a7bdccf0c203e1029a2a93cc2fc8e375c5a056b1d840ab50ddcL194-R194)

cmitzel-ncino committed:
feat: empty commit to run build pipeline on merge

cmitzel-ncino committed:
feat: add 1gp publish action to template

cmitzel-ncino committed:
build: install Node 20 for Semantic Release in workflow template

cmitzel-ncino committed:
feat: add initial project setup with .gitignore, package.json, and publish workflow

abath-ncino commented on workflow-templates/1gp-publish.yml:
```diff
@@ -0,0 +1,293 @@
+ push:
+  branches:
+   - release
+   # - "*.*.x"
```
dead code?

cmitzel-ncino committed:
fix: correct syntax in jq command for selecting default package directory

cmitzel-ncino committed:
fix: remove commented branch pattern from 1gp-publish workflow

Kevmo92 commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,60 @@
+name: 📚 1GP Publish 📝
```
```suggestion
name: 📚 Template publish 📝
```

Kevmo92 commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,60 @@
+            - name: 🦃 Install Node 20 for Semantic Release
+              uses: actions/setup-node@v4
+              with:
+                node-version: 20
```
can we go to 22 here instead?

cmitzel-ncino replied:
🚀

Kevmo92 commented on .gitignore:
```diff
@@ -0,0 +1,11 @@
+### IDE ###
+## VS Code ##
+.vscode/*
+!.vscode/settings.json
```
```suggestion
!.vscode/settings.json

```

Kevmo92 commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,60 @@
+              run: |
+                echo "## ✏️ ✅ Published Release 📦
+                    * [${{ steps.semantic-release.outputs.tag }}](${{ steps.semantic-release.outputs.url }})
+                " >> $GITHUB_STEP_SUMMARY
```
```suggestion
                " >> $GITHUB_STEP_SUMMARY

```

cmitzel-ncino committed:
Update .github/workflows/publish.yml

Co-authored-by: Kevin Watson <Kevmo92@users.noreply.github.com>

cmitzel-ncino committed:
Update .gitignore

Co-authored-by: Kevin Watson <Kevmo92@users.noreply.github.com>

cmitzel-ncino committed:
Update .github/workflows/publish.yml

Co-authored-by: Kevin Watson <Kevmo92@users.noreply.github.com>

cmitzel-ncino committed:
Merge remote-tracking branch 'origin/release' into feat/DO-3489/cmitzel

cmitzel-ncino committed:
refactor: update CI workflows and properties

- Simplified the `1gp-ci.yml` by removing unnecessary branch patterns and added a new `merge_group` trigger.
- Updated the `1gp-publish.properties.json` description for clarity.
- Refined the `1gp-publish.yml` workflow by restructuring jobs and enhancing clarity in job names and steps.
- Adjusted string formatting for consistency across YAML files.
- Improved the `commit-lint.yml` to use a `merge_group` trigger instead of a `push` trigger for better handling of changes requested in pull requests.

cmitzel-ncino committed:
Merge branch 'feat/DO-3489/cmitzel' of https://github.com//ncino/.github into feat/DO-3489/cmitzel

cmitzel-ncino committed:
chore: remove package.json and package-lock.json files

- Deleted `package.json` and `package-lock.json` as part of project cleanup.

cmitzel-ncino committed:
chore: remove .gitignore file

- Deleted the `.gitignore` file as part of project cleanup.

Cerelius committed:
refactor: enhance 1gp-publish workflow

- Updated environment variables for clarity and consistency.
- Restructured job definitions, including renaming and reordering for better readability.
- Added failure notifications for deployment and upload steps to improve error handling.
- Updated dependencies and actions to their latest versions for better performance and security.
- Removed redundant steps and improved overall workflow efficiency.

cmitzel-ncino committed:
chore: remove obsolete 1GP promote workflow files

- Deleted `1gp-promote.properties.json` and `1gp-promote.yml` as they are no longer needed in the project.
- This cleanup helps streamline the workflow templates and reduce clutter.

cmitzel-ncino committed:
Merge branch 'release' into feat/DO-3489/cmitzel

cmitzel-ncino committed:
fix: spacing

Kevmo92 commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,60 @@
+                token: ${{ secrets.CI_USER_REPO_ACCESS_TOKEN }}
+
+            - name: 🛠️ Force Repository Setup
+              uses: ncino/action-force-setup@v3
```
v4 is latest!

Cerelius commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,60 @@
+name: 📚 Template publish 📝
```
Even though this is accurate, I've found that people constantly forget to update the workflow title. Might want to make this 2GP Publish as it would be when completed. 

Or, we use it as a test to see if they can follow instructions 🤷

Cerelius commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,60 @@
+  COMMIT_SHA1: ${{ github.sha }}
+  GIT_BRANCH: ${{ github.ref_name }}
+  REPO_NAME: ${{ github.event.repository.name }}
+  NAMESPACE: 'nCino'
```
Should this be placeholdered?
